sharkfuser/tests/test_graph.cpp

```diff
@@ -98,3 +98,55 @@ TEST_CASE("Graph asm_emitter requires validation to be run first", "[graph]") {
   // ASM emitter should now work
   REQUIRE(isOk(g.emitAsm()));
 }
+
+TEST_CASE("Graph generateCompiledArtifacts and readCompiledArtifacts",
+          "[graph]") {
+  SECTION("Caching behavior for valid input IR") {
+    // Build valid graph.
+    int64_t n = 16, c = 128, h = 64, w = 64, k = 256, r = 1, s = 1;
+    Graph g;
+    g.setName("test_graph");
+    g.setIODataType(DataType::Half).setComputeDataType(DataType::Float);
+    auto X = g.tensor(TensorAttr()
+                          .setName("image")
+                          .setDim({n, c, h, w})
+                          .setStride({c * h * w, h * w, w, 1}));
+    auto W = g.tensor(TensorAttr()
+                          .setName("filter")
+                          .setDim({k, c, r, s})
+                          .setStride({c * r * s, r * s, s, 1}));
+    auto conv = ConvFPropAttr()
+                    .setPadding({0, 0})
+                    .setStride({1, 1})
+                    .setDilation({1, 1})
+                    .setName("conv_fprop");
+    auto Y = g.convFProp(X, W, conv);
+    Y->setDim({n, k, h, w}).setStride({k * h * w, h * w, w, 1});
+    Y->setOutput(true);
+    REQUIRE(isOk(g.validate()));
+
+    // Generate asm.
+    std::string generatedAsm = TEST_FUSILLI_TRY(g.emitAsm());
+
+    // Cache should be empty.
+    REQUIRE(isError(g.readCompiledArtifacts(Backend::CPU, generatedAsm)));
```

Comment:
Let's also test the `g.readOrGenerateCompiledArtifacts` wrapper here once you have it. This will be easier to call from the main `g.execute` method.